allauth/account/migrations/0003_email_max_length.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import models, migrations
+from django.conf import settings
+
+UNIQUE_EMAIL = getattr(settings, 'ACCOUNT_UNIQUE_EMAIL', True)
+EMAIL_MAX_LENGTH = getattr(settings, 'ACCOUNT_EMAIL_MAX_LENGTH', 254)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('account', '0002_email_max_length'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='emailaddress',
+            name='email',
+            field=models.EmailField(unique=UNIQUE_EMAIL, max_length=EMAIL_MAX_LENGTH, verbose_name='e-mail address'),
+
+        ),
+    ]
+
+    if not UNIQUE_EMAIL:
```

Comment:
Instead of creating a new migration, migration `0002_email_max_length` should be altered to simply use the setting over at:

```
            field=models.EmailField(unique=UNIQUE_EMAIL, max_length=254, verbose_name='e-mail address'),
```

This way, most deployments are not impacted or bothered by additional migrations which don't change the database at all. Also, if you were to keep the above migration and upgrade allauth and alter the setting thereafter, things still wouldn't work as the above migration did already run with a default of 254.